oshi-core/src/main/java/oshi/util/platform/windows/WmiUtil.java

```diff
@@ -443,7 +439,7 @@ public static void initCOM() {
             // This can be safely ignored
             if (COMUtils.FAILED(hres) && hres.intValue() != WinError.RPC_E_TOO_LATE) {
                 Ole32.INSTANCE.CoUninitialize();
-                throw new COMException("Failed to initialize security.");
+                throw new COMException("Failed to initialize security.", new HRESULT(-1));
```

Comment:
Better solution for OSHI, definitely, although I still think we should trap the null in the other code.

But why use -1 here when we have a hres variable (already a `HRESULT` type available to us?  The correct fix here (and 12 lines earlier) is just to include `hres` as the second argument. :)

Reply:
Oh ! sorry I didn't see the href variable, I will make the modifications. And at a case if href is null too.